app/controller/panel/TimeSlider.js

```diff
@@ -0,0 +1,170 @@
+
+        if (!startDate || !endDate) {
+            Ext.log.warn('Please provide valid start / end date.');
+            return;
+        }
+
+        var timeRangeObj = me.getTimeRangeForDate(startDate, endDate,
+            selectedDate, timeIncrementUnit);
+
+        slider.setMaxValue(timeRangeObj.maxValue);
+        slider.setMinValue(timeRangeObj.minValue);
+
+        viewModel.setData({
+            timeRangeObj: timeRangeObj
+        });
+    },
+
+    /**
+     * Function called in time slider changes
+     * @param {Ext.slider.Single | Ext.slider.Multi} slider The slider
+     * @param {Number} value The new value of the slider
+     */
+    onTimeChanged: function (slider, value) {
+        // TODO: WFS layers must be filtered
+        var me = this;
+        var layerNames = me.getView().layerNames;
+        Ext.each(layerNames, function (layerName) {
+            var layer = BasiGX.util.Layer.getLayerByName(layerName);
+            if (layer && (layer.getSource() instanceof ol.source.TileWMS || layer.getSource() instanceof ol.source.ImageWMS)) {
+                var wmsLayerSource = layer.getSource();
+                var sliderDate = me.getDateForSliderValue(value);
+                if (wmsLayerSource && sliderDate) {
+                    wmsLayerSource.updateParams({
+                        TIME: Ext.Date.format(sliderDate, 'c')
+                    });
+                }
+            }
+        });
+    },
+
+    /**
+     * Function to generate tooltip text for slider
+     *
+     * @param {Ext.slider.Thumb} thumb The Thumb that the tip is attached to
+     * @return {String} The formatted date as text
+     */
+    getTipText: function (thumb) {
+        var me = this;
+        return me.getDateStringForSliderValue(thumb.value);
+    },
+
+    /**
+     * Function that returns formatted date string vor passed slider value
+     *
+     * @param {Number} sliderValue The value of the slider
+     * @return {String} The formatted date as text
+     */
+    getDateStringForSliderValue: function (sliderValue) {
+        var me = this;
+        var view = me.getView();
+        var timeIncrementUnit = view.timeIncrementUnit || 'year';
+        var dateFormat = '';
+        switch (timeIncrementUnit) {
+        case 'month':
+            dateFormat = 'm/Y';
+            break;
+        default:
+            dateFormat = 'Y';
+        }
+        return Ext.Date.format(
+            me.getDateForSliderValue(sliderValue),
+            dateFormat
+        );
+    },
+
+    /**
+     * Function that returns the date vor passed slider value
+     *
+     * @param {Number} sliderValue The value of the slider
+     * @return {Date} The date
+     */
+    getDateForSliderValue: function (sliderValue) {
+        var me = this;
+        var view = me.getView();
+        var timeIncrementUnit = view.timeIncrementUnit || 'year';
+        switch (timeIncrementUnit) {
+        case 'month':
```

Comment:
I personally do not like the indentation here, but this is a personal thing, I guess.